src/main/java/jarvis/model/Lesson.java

```diff
@@ -81,4 +83,12 @@ public TimePeriod getTimePeriod() {
     public LessonAttendance getAttendance() {
         return attendance;
     }
+
+    public LessonNotes getNotes() {
+        return notes;
+    }
+
+    public abstract String getType();
```

Comment:
I don't think this is necessary. If we do need to check which of the Lesson subclasses an object is, we can use `instanceof`. Alternatively, the lesson subtypes should be an enum instead of a string to avoid introducing bugs if we change it in the future.